File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> elements) {
    int n = elements.size();
    
    vector<int> ex = elements;
    ex.insert(ex.end(), elements.begin(), elements.end());
    
    unordered_set<int> answer;
    for (int i = 0; i < n; i++) {
        int current = 0;
        for (int j = 0; j < n; j++) {
            current += ex[i + j];
            answer.insert(current);
        }
    }
    
    return answer.size();
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    
    ex = elements + elements
    
    answer = set()
    for i in range(n):
        current = 0
        for j in range(n):
            current += ex[i + j]
            answer.add(current)
    
    return len(answer)